flang/lib/Lower/CallInterface.cpp

```diff
@@ -21,11 +22,18 @@
 #include "flang/Semantics/tools.h"
 
 //===----------------------------------------------------------------------===//
-// Caller side interface implementation
+// BIND(C) mangling helpers
 //===----------------------------------------------------------------------===//
 
-bool Fortran::lower::CallerInterface::hasAlternateReturns() const {
-  return procRef.hasAlternateReturns();
+// Remove leading and trailing spaces from a string.
+// Bind(C) NAME= expression may have trailing and leading space that must be
+// removed according to 18.9.2 point 2.
+static std::string adjustBindCName(std::string &&str) {
+  auto first = str.find_first_not_of(" ");
+  auto last = str.find_last_not_of(" ");
+  if (first == std::string::npos || last == std::string::npos)
+    llvm::report_fatal_error("Name in Bind(C) has no non-space character");
```

Comment:
I think it would make more sense to strip the spaces when we set the bindName during semantics so that it is correct in the symbol table. Indeed, it should probably be stored as a string rather than an expression. If the name only has spaces it looks like it should be treated as if the binding name weren't there at all. It would be nice to warn about it though.

All of that can happen later; it doesn't need to affect this review.

Reply:
I've created a patch to the front end to save the correct binding label in symbols as a string: https://reviews.llvm.org/D99208

After that change and this one are merged, I'll create a PR here to bring it in. That should simplify the code.

Reply:
Thanks, dealing with this in the font-end will be much nicer.